### PR TITLE
Tweak LDFLAGS in Makefile

### PR DESCRIPTION
**Pull request by jimmidyson**
**What this PR does / why we need it**:

This PR tweaks the use of LDFLAGS in the Makefile to allow overriding LDFLAGS at the time of running `make`, while still retaining the necessary version linker flags. This was a problem in helm package in Archlinux, which overrides linker flags with:

```
make LDFLAGS="-s -w -linkmode external" GOFLAGS="-buildmode=pie -trimpath"
```

Due to the nature of make variables, this meant that the necessary version linker flags do not include the version linker flags, leading to `helm version` being incorrect:

```
$ helm version
version.BuildInfo{Version:"v3.3", GitCommit:"", GitTreeState:"", GoVersion:"go1.15.2"}
```

After these changes, the helm binary is correctly built and reports version correctly:

```
$ helm version
version.BuildInfo{Version:"v3.3.1", GitCommit:"249e5215cde0c3fa72e27eb7a30e8d55c9696144", GitTreeState:"clean", GoVersion:"go1.15.2"}
```

This PR also consistently uses parentheses for make variables (parentheses and braces are interchangeable in a Makefile so this is just for tidiness, not functionality).

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by jimmidyson** [comment]
@Foxboron FYI - I've seen your recent PR about this and wonder if you could review this please?

**Comment by Foxboron** [comment]
Hmm, I think this is cleaner in the long run as you get more control in the build system if you need it without sacrificing current usage . Admittedly the `$EXT_LDFLAGS` patch was the easier option then reworking the `$LDFLAGS` handling.

**Comment by jimmidyson** [comment]
@bacongobbler I saw that you reviewed the previous PR that touched linker flags - would you mind taking a look at this one too please?

File: Makefile [hunk 1]
```diff
@@ -32,25 +32,24 @@ GIT_TAG    = $(shell git describe --tags --abbrev=0 --exact-match 2>/dev/null)
...
+BINARY_VERSION = $(VERSION)
+else
+BINARY_VERSION = $(GIT_TAG)
 endif
-BINARY_VERSION ?= ${GIT_TAG}
 
 # Only set Version if building a tag or VERSION is set
 ifneq ($(BINARY_VERSION),)
-	LDFLAGS += -X helm.sh/helm/v3/internal/version.version=${BINARY_VERSION}
+VERSION_LDFLAGS += -X helm.sh/helm/v3/internal/version.version=$(BINARY_VERSION)
```
Comment (mattfarina) [inline Makefile:42]: Out of curiosity, why did you remove the indentation of the body of the conditionals?

**Review by mattfarina — CHANGES_REQUESTED** [review]

**Review by mattfarina — CHANGES_REQUESTED** [review]
Thanks for the PR and for making this easier to build in other environments.

Aside from my one question, this PR changes the `build-cross` make target which accepts the `LDFLAGS`. For example, the EXT_LDFLAGS is no longer part of it. The `build-cross` target should be updated in a similar manner to the way the `build` target was updated.

File: Makefile (hunk 1 above)
Comment (jimmidyson) [reply Makefile:42]: I believe it's more idiomatic to not indent conditionals - see https://www.gnu.org/software/make/manual/html_node/Conditional-Syntax.html for examples that show no indentation.

**Comment by jimmidyson** [comment]
@mattfarina Good spot about the `build-cross` target - thanks! Updated and pushed.

**Comment by jimmidyson** [comment]
@mattfarina Is there anything else in this PR that needs addressing?

File: Makefile (hunk 1 above)
Comment (mattfarina) [reply Makefile:42]: The conditional examples in the manual have an indent. See https://www.gnu.org/software/make/manual/html_node/Conditional-Example.html#Conditional-Example. I was looking at those two pages before I made my comment because I wanted to find out what was idiomatic. I couldn't tell because those two pages in the manual aren't consistent.

Do you have another piece of information to show that it's idiomatic to not indent? If not, to be consistent with what we have had I would ask to keep the indent. It's a little easier to read.

Comment (jimmidyson) [reply Makefile]: @mattfarina Fair point. I've reverted the indentation.

**Comment by mattfarina** [comment]
I'm curious why you couldn't use the existing `EXT_LDFLAGS` to pass in `-linkmode external`. The `-s -w` was already set by default.

Is there something I'm missing? It could entirely be the case.

**Comment by jimmidyson** [comment]
Hi @mattfarina sorry for late reply. My reasoning is in https://github.com/helm/helm/pull/8746#issuecomment-694154281. If you don't feel that is sufficient reason, then let's close this pr, no worries.
